### PR TITLE
feat(datastore-lock)!: factoring out datastore-lock

### PR DESCRIPTION
**Pull request by tmatsuo**
This small library is initially implemented in `blunderbuss`, but it should be useful for some other bots, so I'm trying to make it a npm package.

### CODE REVIEW

**Review by chingor13 — APPROVED** [review]
Will want @bcoe's feedback on naming. We can set up the release config in a followup PR.

File: packages/datastore-lock/src/datastore-lock.ts [hunk 1]
```diff
@@ -0,0 +1,171 @@
...
+      waitTime = Math.min(
+        maxDelay,
+        waitTime + Math.floor(Math.random() * 2000)
+      );
+    }
+    // Timeout
+    return false;
+  }
+
+  private async _acquire(): Promise<AcquireResult> {
```
Comment (bcoe) [inline packages/datastore-lock/src/datastore-lock.ts:105]: Fun fact, I learned recently that `private` fields aren't actually private outside of the context of TypeScript, so when I actually wanted private methods, I ended up using symbols as method names:

https://github.com/yargs/yargs/blob/master/lib/yargs-factory.ts#L88

I'm not suggesting we change anything, just wanted to share this surprising TypeScript behavior.

Comment (tmatsuo) [reply packages/datastore-lock/src/datastore-lock.ts:105]: Ah it makes sense.

**Review by tmatsuo — COMMENTED** [review]
Changed the package name to `@github-automations/datastore-lock`.
@bcoe PTAL

**Comment by tmatsuo** [comment]
I'll add release config first :)

**Comment by tmatsuo** [comment]
I'm trying to rewrite the first commit message to be a breaking change, but my local git client drops the exclamation mark.

I'm going to squash locally and re-push the change with a new commit message.

**Review by tmatsuo — COMMENTED** [review]
@bcoe I guess I'm ready for the release, PTAL

**Pull request merged**